packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -40,8 +45,16 @@ export function isTreeNodeSchemaClass<
  * Types for use in fields.
  * @remarks
  * Type constraint used in schema declaration APIs.
+ *
+ * The order of types in the array is not significant.
+ * Additionally it is legal for users of this type to have the runtime and compile time order of items withing this array not match.
+ * Therefor to ensure type safety, this these arrays should not be indexed, and instead just be iterated.
+ *
+ * Ideally this restriction would be modeled in the type itself, but it is not ergonomic to do so as there is no easy (when compared to arrays)
+ * way to declare and manipulate unordered sets of types in TypeScript.
+ *
  * Not intended for direct use outside of package.
- * @public
+ * @system @public
```

Comment:
This is also technically breaking. How confident are we that customers are not directly referencing this type in some way?

Reply:
Since we have the comment "Not intended for direct use outside of package." no usage if this was possible if using as documented, so I think its ok. As that comment landed just before 2.0 (when system wasn't a well documented thing), I think this is fine.